### PR TITLE
Add self-signed ca script for development

### LINKED ISSUE #642: [help wanted] KFServing webhook cert installation script
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
https://github.com/kubeflow/kfserving/pull/630 added cert manager to provision webhook certs and this is suggested way for production grade installation, however we still need a way for deploying KFServing for development or for users who does not want this dependency. 

**Anything else you would like to add:**
Looks like we can leverage this script(https://github.com/kubeflow/manifests/blob/master/admission-webhook/bootstrap/base/config-map.yaml#L9) in kubeflow manifest to generate self signed cert and patch to the validation/mutating webhook configuration.

### PR DESCRIPTION
**Pull request by Tomcli**
**What this PR does / why we need it**:
Add a self-signed ca script for development without cert-manager.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #642

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/650)

### CODE REVIEW

File: docs/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -140,6 +140,12 @@ You can follow
 [the cert manager documentation](https://docs.cert-manager.io/en/latest/getting-started/install/kubernetes.html)
 to install it.
 
+If you don't want to install cert manager, you can set the `SELF_SIGNED_CA` environment variable to true.
+`SELF_SIGNED_CA` will execute a script to create a self-signed CA and patch it to the webhook config.
+```bash
+export SELF_SIGNED_CA=true
```
Comment (ellis-bigelow) [inline docs/DEVELOPER_GUIDE.md]: nit: KFSERVING_ENABLE_SELF_SIGNED_CA

**Comment by ellis-bigelow** [comment]
Very cool. Thank you for your contribution!

File: docs/DEVELOPER_GUIDE.md (hunk 1 above)
Comment (Tomcli) [reply docs/DEVELOPER_GUIDE.md]: Thanks ellis.

**Comment by yuzisun** [comment]
Awesome! Thanks @Tomcli !

/approve

**Pull request merged**